### PR TITLE
adds basic components and hooks header up to router

### PR DESCRIPTION
**Pull request by mcraepv**
Also adds bootstrap, react router dom, etc

### CODE REVIEW

**Review by aaronpaulroberts — CHANGES_REQUESTED** [review]
Please change all of your functions to arrow functions.
ex:
- With logic
`const Header = () => {
 ...logic...
  return(<jsx>)
}`

- no logic
` const Header = () => (<jsx>)`

File: frontend/.eslintcache [hunk 1]
```diff
@@ -1 +1 @@
...
...",[],"/Users/mcraepv/projects/apprentice-react-o-shop/frontend/src/pages/Home/Home.js",[],"/Users/mcraepv/projects/apprentice-react-o-shop/frontend/src/pages/CheckOut/CheckOut.js",[],"/Users/mcraepv/projects/apprentice-react-o-shop/frontend/src/utils/history.js",["60"],"/Users/mcraepv/projects/apprentice-react-o-shop/frontend/src/pages/Products/Products.js",[],"/Users/mcraepv/projects/apprentice-react-o-shop/frontend/src/pages/Orders/Orders.js",[],"/Users/mcraepv/projects/apprentice-react-o-shop/frontend/src/pages/OrderSuccess/OrderSuccess.js",[],"/Users/mcraepv/projects/apprentice-react-o-shop/frontend/src/pages/AdminOrders/AdminOrders.js",[],{"ruleId":"61","replacedBy":"62"},{"ruleId":"63","replacedBy":"64"},{"ruleId":"61","replacedBy":"65"},{"ruleId":"63","replacedBy":"66"},{"ruleId":"67","severity":2,"message":"68","line":3,"column":16,"nodeType":"69","endLine":3,"endColumn":26},"no-native-reassign",["70"],"no-negated-in-lhs",["71"],["70"],["71"],"react-hooks/rules-of-hooks","React Hook \"useHistory\" cannot be called at the top level. React Hooks must be called in a React function component or a custom React Hook function.","Identifier","no-global-assign","no-unsafe-negation"]
```
Comment (aaronpaulroberts) [inline frontend/.eslintcache]: I think this file, and all files, require an extra line at the end of the file.

File: frontend/src/components/Header/Header.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
+import { React, useState } from 'react';
+import './Header.scss';
```
Comment (aaronpaulroberts) [inline frontend/src/components/Header/Header.js]: is there any particular reason why you are not using css modules?

https://create-react-app.dev/docs/adding-a-css-modules-stylesheet/

File: frontend/src/components/Main/Main.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import Home from '../../pages/Home/Home';
+import Products from '../../pages/Products/Products';
+import AdminOrders from '../../pages/AdminOrders/AdminOrders';
+import AdminProducts from '../../pages/AdminProducts/AdminProducts';
+import CheckOut from '../../pages/CheckOut/CheckOut';
+import Orders from '../../pages/Orders/Orders';
+import OrderSuccess from '../../pages/OrderSuccess/OrderSuccess';
+import ShoppingCart from '../../pages/ShoppingCart/ShoppingCart';
+
+function Main() {
```
Comment (aaronpaulroberts) [inline frontend/src/components/Main/Main.js]: I would recommend putting all of the string path names into a global constant and replace the uses of those strings with their constant reference

File: frontend/src/components/Header/Header.js (hunk 2 above)
Comment (mcraepv) [reply frontend/src/components/Header/Header.js]: This was how the worksheet told us to do it, I believe. I can change it thought if thats the preferred way

File: frontend/.eslintcache (hunk 1 above)
Comment (mcraepv) [reply frontend/.eslintcache]: Added extra line and checked other files for extra line as well

File: frontend/src/components/Main/Main.js (hunk 3 above)
Comment (mcraepv) [reply frontend/src/components/Main/Main.js]: Will do

**Review by aaronpaulroberts — APPROVED** [review]
Good work!

**Pull request merged**